Improve the description of splits and fees (#596)
Podcastindex-org/podcast-namespace: value/value.md

### Payment calculation

A given recipient will receive
```

> (Payment amount) * (Interval payout) * (Interval count)

Reviewer: Can payment amount, interval payout, and interval count be defined here? I assume payment amount is the user's percentage of the total payment amount and interval payout * count is the total payment amount.

Author: Good catch! That part should be adjusted.

Author: What do you think?

https://github.com/Podcastindex-org/podcast-namespace/pull/596/commits/87284605b6dfe2c7446d027748db5887e661b585

Reviewer: Is it worth making a distinction between intervaled and non-intervaled payments? Is the per-recipient calculation different for them?

Author: The calculation is not different, it's just that `Interval count` may not make sense for boosts, so I thought it needs to be at least acknowledged.

I'm not really sure about that part. Open to suggestions on how to better word it. Or if it's necessary at all.

Reviewer 2: @MerryOscar I agree the `split` shouldn't be a percentage, but should have the percentage derived from the `split total`. 
However, what we need a consensus on is how to calculate the `split total`, because some may add all splits including `fee=true` and others may use just the `fee=false` splits. I'm in favor of simply adding all of the splits including 'fee=true' to come up with the total split, but that's not a hill I'm willing to die on. 

My reason for adding all of the splits is it prevents this scenario.  Imagine the 'fee=false' are `a: 50` and `b: 50`. So, the `split total` so far is 100. Now, imagine someone puts in their podcast a 'fee=true' and the split is 200. How is that handled for a 1000 sat boost? By adding all of the splits, including `fee=true`, this becomes less problematic.

A hill I am willing to die on is I won't ever **add** `fees` to the boost. When I boost 1000 sats, I expect 1000 sats to be taken out of my wallet, and would be bothered if I saw 1200 sats removed because the podcaster had a bunch of services adding up to 20%. That 20% should come out of the podcasters pocket not mine. And in the case above, where fees add up to 200%, the podcaster would end up sending 3000 sats even though they only intended to send 1000 sats.

The only caveat to that would be if the app decides to take a cut. I could see an argument for adding the app's cut to the top of the boost, almost like a tax, since the podcaster didn't decide which app the listener would use, and one app could decide to take a 20% cut if they wanted to. In that case, the app shouldn't get 20% of what was intended for the podcaster, because the podcaster didn't have a say in that. However, with the splits, the podcaster has complete control of the split amounts, and should be responsible for paying them out of the listeners initial boost amount.

### Payment calculation

When all combined, together the recipients will receive
```

> (Batch payout) = (Interval count) * (Unit amount)

Reviewer: ```suggestion
(Batch payout) = (Listener payment) * (Interval count)
```

> When all combined, together the recipients will receive
> ```
> (Batch payout) = (Interval count) * (Unit amount)
> ```
> 
> `Unit amount` refers to one of the following
> - either the amount of a one-time payment (with `Interval count` usually being 1)
> - or the amount being streamed per minute (with `Interval count` referring to the interval in minutes at which the payouts are made)
> 
> How `Batch payout` is divided between the recipients depends on their `split` and `fee` values.

Reviewer: ```suggestion
When all combined, together the recipients will receive
```
(Batch payout) = (Listener payment) * (Interval count)
```

`Listener payment` and `Interval count` refers to one of the following scenarios:
- either the amount of a one-time payment made by the listener (with `Interval count` usually being 1)
- or the amount being streamed per-minute by the listener (with `Interval count` referring to the interval in minutes at which payments are sent by their client)

How `Batch payout` is divided between the recipients depends on their `split` and `fee` values.
```

> `Unit amount` refers to one of the following
> - either the amount of a one-time payment (with `Interval count` usually being 1)
> - or the amount being streamed per minute (with `Interval count` referring to the interval in minutes at which the payouts are made)
> 
> How `Batch payout` is divided between the recipients depends on their `split` and `fee` values.

Reviewer: ```suggestion
`Listener payment` and `Interval count` refers to one of the following scenarios:
- either the amount of a one-time payment made by the listener (with `Interval count` usually being 1)
- or the amount being streamed per-minute by the listener (with `Interval count` referring to the interval in minutes at which payments are sent by their client)

How `Batch payout` is divided between the recipients depends on their `split` and `fee` values.
```

> For `fee=false` recipients $X_1, X_2, \ldots, X_m$ with splits $x_1, x_2, \ldots, x_m$ and `fee=true` recipients $Y_1, Y_2, \ldots, Y_n$ with splits $y_1, y_2, \ldots, y_n$:

Reviewer: Make X first since fee payments are first?
```suggestion
For `fee=true` recipients $X_1, X_2, \ldots, X_m$ with splits $x_1, x_2, \ldots, x_m$ and `fee=false` recipients $Y_1, Y_2, \ldots, Y_n$ with splits $y_1, y_2, \ldots, y_n$:
```

> 1. Firstly, `fee=true` recipients $Y_1, Y_2, \ldots, Y_n$ will receive $y_1\\%$, $y_2\\%$, $\ldots$, $y_n\\%$ of the total batch payout.
> 2. Then, the remaining amount will be distributed among `fee=false` recipients $X_1, X_2, \ldots, X_m$ in the ratio $x_1 : x_2 : \ldots : x_m$.

Reviewer: ```suggestion
1. Firstly, `fee=true` recipients $X_1, X_2, \ldots, X_n$ will receive $x_1\\%$, $x_2\\%$, $\ldots$, $x_n\\%$ of the total batch payout.
2. Then, the remaining amount will be distributed among `fee=false` recipients $Y_1, Y_2, \ldots, Y_m$ in the ratio $y_1 : y_2 : \ldots : y_m$.
```

#### Example 3: recipients with `fee=true`

```xml

<podcast:value type="lightning" method="keysend" suggested="0.00000015000">
    <podcast:valueRecipient
            name="A"
            type="node"
            address="02d5c1bf8b940dc9cadca86d1b0a3c37fbe39cee4c7e839e33bef9174531d27f52"
            split="101"
    />
    <podcast:valueRecipient
            name="B"
            type="node"
            address="032f4ffbbafffbe51726ad3c164a3d0d37ec27bc67b29a159b0f49ae8ac21b8508"
            split="99"
    />
    <podcast:valueRecipient
            name="C"
            type="node"
            address="03ae9f91a0cb8ff43840e3c322c4c61f019d8c1c3cea15a25cfc425ac605e61a4a"
            split="16"
            fee="true"
    />
    <podcast:valueRecipient
            name="D"
            type="node"
            address="039e2bd74dadb4a50c3be632a5068cbc820d520635acc5c527a2171ab6cab43a9c"
            split="4"
            fee="true"
    />
</podcast:value>
```

In this case, there are two `fee=true` recipients, so the calculation will have to be done in two steps:

1. `C` will receive $16\\%$ and `D` will receive $4\\%$ off the top of each payment.

> 2. The rest of the payment (i.e., $100\\% - 16\\% - 4\\% = 80\\%$) will be distributed among `A` and `B` in the ratio $101 : 99$.

Reviewer 3: ```suggestion
2. The rest of the payment (i.e., $100\\% - (16\\% + 4\\%) = 80\\%$) will be distributed among `A` and `B` in the ratio $101 : 99$.
```
Just to help clarify that fee percentages are deducted from the total in subsequent steps (as per my question in the linked issue).
